src/elements/content-sidebar/DocGenSidebar/DocGenSidebar.tsx

```diff
@@ -84,49 +86,60 @@ const DocGenSidebar = ({ getDocGenTags }: Props) => {
         });
 
         return jsonPathsMap;
-    };
+    }, []);
 
-    const loadTags = async () => {
-        setIsLoading(true);
-        try {
-            const response: DocGenTemplateTagsResponse = await getDocGenTags();
-            if (response && !!response.data) {
-                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-                // @ts-ignore
-                const { data } = response || [];
-
-                // anything that is not an image tag for this view is treated as a text tag
-                const textTags = data?.filter(tag => tag.tag_type !== 'image') || [];
-                const imageTags = data?.filter(tag => tag.tag_type === 'image') || [];
-                setTags({
-                    text: textTags,
-                    image: imageTags,
-                });
-                setJsonPaths({
-                    textTree: tagsToJsonPaths(textTags),
-                    imageTree: tagsToJsonPaths(imageTags),
-                });
-                setHasError(false);
-            } else {
+    const loadTags = useCallback(
+        async (attempts = 10) => {
+            if (attempts <= 0) {
+                setIsLoading(false);
+                return;
+            }
+            setIsLoading(true);
+            try {
+                const response: DocGenTemplateTagsResponse = await getDocGenTags();
+                if (response.message) {
```

Comment:
Do we need to check for a null response here like we're doing in the else if statement below. Perhaps something like if(!!response.?message) ?